src/Component.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\ProcessorIconv;
+
+use Keboola\Component\BaseComponent;
+use Keboola\Component\UserException;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Finder\Finder;
+
+class Component extends BaseComponent
+{
+    public function run(): void
+    {
+        /** @var Config $config */
+        $config = $this->getConfig();
+        try {
+            iconv($config->getSourceEncoding(), 'UTF-8', 'abc');
+        } /** @noinspection PhpRedundantCatchClauseInspection */ catch (\ErrorException $e) {
+            throw new UserException('Source encoding ' . $config->getSourceEncoding() . ' is invalid.');
+        }
+        try {
+            iconv('UTF-8', $config->getTargetEncoding(), 'abc');
+        } /** @noinspection PhpRedundantCatchClauseInspection */ catch (\ErrorException $e) {
+            throw new UserException('Target encoding ' . $config->getTargetEncoding() . ' is invalid.');
+        }
+        $filter = 'convert.iconv.' . $config->getSourceEncoding() .  '/' .
+            $config->getTargetEncoding() . $config->getModifier();
+
+        $finder = new Finder();
+        $finder->in($this->getDataDir() . '/in/tables')->files();
+        $this->processDir($finder, 'tables', $filter);
+        $finder = new Finder();
+        $finder->in($this->getDataDir() . '/in/files')->files();
+        $this->processDir($finder, 'files', $filter);
+    }
+
+    private function processDir(Finder $finder, string $dir, string $filter) : void
+    {
+        $fs = new Filesystem();
+        foreach ($finder as $inFile) {
+            if ($inFile->getExtension() == 'manifest') {
+                // copy manifest without modification
+                $fs->copy($inFile->getPathname(), $this->getDataDir() . "/out/$dir/" . $inFile->getFilename());
```

Comment:
zalomit prosím

Reply:
Proto říkám "prosím" :)

Ale souhlas - co si nestěžuje standard je správně, dokud není standard, kde to je špatně :)